internal/pkg/generator/ecs_service.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package generator generates a command given an ECS service or a workload.
+package generator
+
+import (
+	"fmt"
+
+	"github.com/aws/copilot-cli/internal/pkg/aws/ecs"
+
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+type ecsServiceGetter interface {
+	Service(clusterName, serviceName string) (*ecs.Service, error)
+	TaskDefinition(taskDefName string) (*ecs.TaskDefinition, error)
+	NetworkConfiguration(cluster, serviceName string) (*ecs.NetworkConfiguration, error)
+}
+
+// ECSServiceCommandGenerator generates task run command given an ECS service.
+type ECSServiceCommandGenerator struct {
+	Cluster          string
+	Service          string
+	ECSServiceGetter ecsServiceGetter
+}
+
+// Generate generates a task run command.
+func (g ECSServiceCommandGenerator) Generate() (*GenerateCommandOpts, error) {
+	networkConfig, err := g.ECSServiceGetter.NetworkConfiguration(g.Cluster, g.Service)
+	if err != nil {
+		return nil, fmt.Errorf("retrieve network configuration for service %s in cluster %s: %w", g.Service, g.Cluster, err)
+	}
+
+	svc, err := g.ECSServiceGetter.Service(g.Cluster, g.Service)
+	if err != nil {
+		return nil, fmt.Errorf("retrieve service %s in cluster %s: %w", g.Service, g.Cluster, err)
+	}
+
+	taskDefNameOrARN := aws.StringValue(svc.TaskDefinition)
+	taskDef, err := g.ECSServiceGetter.TaskDefinition(taskDefNameOrARN)
+	if err != nil {
+		return nil, fmt.Errorf("retrieve task definition %s: %w", taskDefNameOrARN, err)
+	}
+
+	if len(taskDef.ContainerDefinitions) > 1 {
+		return nil, fmt.Errorf("found more that one container in task definition: %s", taskDefNameOrARN)
```

Comment:
```suggestion
		return nil, fmt.Errorf("found more than one container in task definition %s", taskDefNameOrARN)
```

Reply:
If it could take the name of service other than those created by copilot, do we need to validate it has at least one container as well?

Reply:
It's not possible to create a task definition with zero container. 

But it doesn't hurt to validate either 😆 so I will just do that since it doesn't seem to be explicitly documented anywhere that task def will have >= 1 containers.

Reply:
haha cool please ignore this comment